Burn cell fails to compile with USE_CXX_EOS=FALSE

To reproduce compile `unit_test/burn_cell` with `make -j USE_CXX_EOS=FALSE` 

it gives a long list of errors related to the c++ namespace of helmholtz. It seems like the problem is not actually set up for a fortran eos?
```
/home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../EOS/helmholtz/actual_eos.H:153: undefined reference to `helmholtz::tlo'
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../EOS/helmholtz/actual_eos.H:153: undefined reference to `helmholtz::tstpi'
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../EOS/helmholtz/actual_eos.H:154: undefined reference to `helmholtz::jtmax'
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../EOS/helmholtz/actual_eos.H:155: undefined reference to `helmholtz::dlo'
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../EOS/helmholtz/actual_eos.H:155: undefined reference to `helmholtz::dstpi'
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../EOS/helmholtz/actual_eos.H:156: undefined reference to `helmholtz::itmax'
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../EOS/helmholtz/actual_eos.H:161: undefined reference to `helmholtz::f'
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../EOS/helmholtz/actual_eos.H:161: undefined reference to `helmholtz::f'
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../EOS/helmholtz/actual_eos.H:161: undefined reference to `helmholtz::f'
...
```

and 

```
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../rates/aprox_rates.H:53: undefined reference to `rv'
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../rates/aprox_rates.H:53: undefined reference to `rv'
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../rates/aprox_rates.H:53: undefined reference to `rv'
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../rates/aprox_rates.H:53: undefined reference to `rfdm'
/usr/bin/ld: /home/cdegrendele/codes/Microphysics/unit_test/burn_cell/../../rates/aprox_rates.H:54: undefined reference to `rfd0'
...
```

--- a/EOS/Make.package
+++ b/EOS/Make.package
@@ -1,2 +1,4 @@
 F90EXE_sources += composition.F90
+ifeq ($(USE_CXX_EOS),TRUE)
 CEXE_headers += eos_composition.H
+endif

--- a/interfaces/Make.package
+++ b/interfaces/Make.package
@@ -6,6 +6,7 @@ F90EXE_sources += eos.F90
 F90EXE_sources += eos_override.F90
 F90EXE_sources += eos_type.F90
 
+ifeq ($(USE_CXX_EOS),TRUE)
 CEXE_headers += eos.H
 CEXE_headers += eos_data.H
 CEXE_headers += eos_type.H
@@ -13,6 +14,7 @@ CEXE_headers += eos_override.H
 FEXE_headers += eos_F.H
 
 CEXE_sources += eos_data.cpp
+endif
 
 CEXE_headers += network.H
 ifeq ($(USE_CXX_REACTIONS),TRUE)

--- a/unit_test/burn_cell/main.cpp
+++ b/unit_test/burn_cell/main.cpp
@@ -7,8 +7,8 @@
 using namespace amrex;
 
 #include <extern_parameters.H>
-#include <eos.H>
 #ifdef CXX_REACTIONS
+#include <eos.H>
 #include <network.H>
 #include "burn_cell.H"
 #endif
@@ -51,10 +51,11 @@ int main(int argc, char *argv[]) {
   // Copy extern parameters from Fortran to C++
   init_extern_parameters();
 
+#ifdef CXX_REACTIONS
+
   // C++ EOS initialization (must be done after Fortran eos_init and init_extern_parameters)
   eos_init();
 
-#ifdef CXX_REACTIONS
   // C++ Network, RHS, screening, rates initialization
   network_init();
 #endif

